packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -116,8 +117,27 @@ const VeniaAdapter = props => {
  * other modules in the codebase need access to it.
  */
 VeniaAdapter.apolloLink = apiBase => {
+    // Intercept and shrink URLs from GET queries. Using
+    // GET makes it possible to use edge caching in Magento
+    // Cloud, but risks exceeding URL limits with default usage
+    // of Apollo's http link. `shrinkGETQuery` encodes the URL
+    // in a more efficient way.
+    const customFetchToShrinkQuery = (uri, options) => {
+        let url = uri;
+        if (options.method === 'GET') {
+            url = shrinkGETQuery(uri);
+        }
+        return fetch(url, options);
+    };
+
     return createHttpLink({
-        uri: apiBase
+        uri: apiBase,
+        fetch: customFetchToShrinkQuery,
+        // Warning: useGETForQueries risks exceeding URL length limits. These limits
+        // in practice are typically set at or behind where TLS terminates. For Magento
+        // Cloud and Fastly, 8kb is the maximum by default
+        // https://docs.fastly.com/en/guides/resource-limits#request-and-response-limits
```

Comment:
If the query ends up exceeding this character/byte limit, do we have anything prepared to throw an informative error?

Reply:
Any chance to only remove the offending include statement?

Ie
`meta_title @include(if: $onServer)`
To
`meta_title`

Reply:
We don't, but I do think it's worth consideration.

My biggest fear was that this can be deployed to non-Magento-owned infrastructure, and there will be different limits (some allow > 8kb).

We could always make it configurable and set a default of 8kb (or ∞)

Reply:
> Any chance to only remove the offending include statement?

To clarify @fooman, you still want the data in the default query, just without the conditional directive, right? We could do that. @jimbo thoughts?

Reply:
@DrewML yes - this seems the less intrusive change. Currently I am not sure how the `@include` directive is supposed to work but from the comments so far the idea seems to have been that the query only executes for SSR. However when I browse on a current PR (other than this), meta_title and meta_description do update when navigating the site client side. Keeping the data in the query would not change this behaviour while fixing the issue with `@include`

Reply:
The downside of keeping the fields is that they aren't actually used by anything in pwa studio right now and is both extra data requested and mental overhead when we try to make decisions about new queries or old query maintenance.

I'd rather we provide, for users of pwa-studio, explicit queries that request only the data necessary but that we also allow an easy path to customizing/overriding these queries.

Reply:
> So I'm on board with the less intrusive change of just removing the directives rather than the fields, but I'd like to add a comment block around each field indicating that the data isn't actually used by any of the venia components by default.


Is this something I should push up? Happy to make the change, just want to make sure there's consensus 😃

Reply:
Fields added back _without_ the `@include` directive